lib/src/cpp/utils/ParseXML.cpp

```diff
@@ -260,25 +264,53 @@ xercesc::DOMNode* ParseXML::get_node(
                                       tag_name + " does not have the attribute " + att_name + "\n");
                         }
                     }
-                    // LIST element TODO refactor
-                    else if (&index == nullptr)
+                    // LIST element
+                    else if (index != nullptr)
                     {
-                        int32_t realIndex = index;
-                        // Transform index to real index
-                        if (index < 0)
+                        // Check value of index
+                        std::string in = *index;
+
+                        // Empty index
+                        if (in == "")
                         {
-                            realIndex = node_tag_list->getLength() + index;
+                            // return parent
+                            return tag_node;
                         }
-                        // Check bounds
-                        if (realIndex < 0 || realIndex >= node_tag_list->getLength())
+                        // Index has a value
+                        else
                         {
-                            // Throw eprosima::qosprof::ElementNotFound exception
-                            throw ElementNotFound(
-                                      tag_name + " does not have an element in position " + std::to_string(
-                                          realIndex) + "\n");
+                            // Parse index value and obtain node
+                            int32_t int_index = 0;
+                            int32_t real_index = 0;
+
+                            try
+                            {
+                                // Parse index from string to int
+                                int_index = std::stoi(in);
+                                real_index = int_index;
+                            }
+                            catch (...)
```

Comment:
Instead of catching any exception, I think we should catch the std::exception thrown by [std::stoi](https://en.cppreference.com/w/cpp/string/basic_string/stol). It seems that 2 exceptions can be throw: `invalid_argument` and `out_of_range`. Both of them are children of std::exception, so as we are probably interested on those two exceptions, we should capture the parent one.